packages/core/src/types.ts

```diff
@@ -852,7 +852,7 @@ export interface MiddlewareConfig {
 export interface ApiClientFactoryParams<T, F = any> {
   api: F;
   isProxy?: boolean;
-  onCreate: (config: T, headers?: Record<string, string>) => { config: T; client: any };
+  onCreate: (config: T) => Promise<{ config: T; client: any }>;
```

Comment:
Why are the `headers` removed here?

Reply:
@WojtekTheWebDev look here: https://github.com/vuestorefront/vue-storefront/pull/6807#discussion_r977422605

Reply:
Can we resolve this?

Reply:
Thx for explaining